### PR TITLE
Do not do fs resize on read-only mount

### PR DESCRIPTION
**Pull request by gnufied**
We should not perform file system resize when volume is mounted in read-only mode.

Fixes : https://github.com/kubernetes/kubernetes/issues/56588

```release-note
Do not do file system resize on read-only mounts
```

### CODE REVIEW

**Comment by dixudx** [comment]
@gnufied Please squash your commits. Then lgtm.

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 1]
```diff
@@ -1272,6 +1280,7 @@ func updatePVCStatusCapacity(pvcName string, pvc *v1.PersistentVolumeClaim, capa
 	}
 
 	pvcCopy.Status.Capacity = capacity
+	pvcCopy.Status.Conditions = []v1.PersistentVolumeClaimCondition{}
```
Comment (dixudx) [inline pkg/volume/util/operationexecutor/operation_generator.go]: This is useless?

Comment (gnufied) [reply pkg/volume/util/operationexecutor/operation_generator.go]: This is not useless. This removes conditions set on PVC after resizing has been successful and that is why it exists as separate commit as well.

**Comment by childsb** [comment]
This is bug fix

**Pull request merged**